File: examples/android/CMeSdkJava/app/src/main/java/msdk/java/sample/db/dao/ActionDao.java
package msdk.java.sample.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import msdk.java.sample.db.entity.Action;

@Dao
public interface ActionDao {
    @Query("SELECT * FROM `action`")
    List<Action> getAllAsync();

    @Query("SELECT * FROM `action`")
    LiveData<List<Action>> getAll();

    @Query("SELECT * FROM `action` WHERE id = :id")
    Action getActionsById(int id);

    @Query("SELECT * FROM `action` WHERE status = :status")
    LiveData<List<Action>> getActionsByStatus(String status);

    @Insert
    void insertAll(Action... actions);

    @Update
    void update(Action action);

    @Delete
    void delete(Action action);
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/db/dao/ActionDao.kt
package me.connect.sdk.java.samplekt.db.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import me.connect.sdk.java.samplekt.db.entity.Action

@Dao
interface ActionDao {
    @Query("SELECT * FROM `action`")
    fun getAllAsync(): List<Action>

    @Query("SELECT * FROM `action`")
    fun getAll(): LiveData<List<Action>>

    @Query("SELECT * FROM `action` WHERE id = :id")
    fun getActionsById(id: Int): Action

    @Query("SELECT * FROM `action` WHERE status = :status")
    fun getActionsByStatus(status: String?): LiveData<List<Action>>

    @Insert
    suspend fun insertAll(vararg actions: Action)

    @Update
    suspend fun update(action: Action)

    @Delete
    suspend fun delete(action: Action)
}